di_website/publications/models.py

```diff
@@ -486,16 +485,26 @@ def save(self, *args, **kwargs):
 
     def get_context(self, request, *args, **kwargs):
         context = super().get_context(request, *args, **kwargs)
-
-        context['related_pages'] = get_related_pages(
-            self, self.publication_related_links.all(), PublicationPage.objects)
+        related_options = request.GET.get('choices', None)
+
+        if related_options == 'Topic':
+            queryset = PublicationPage.objects.filter(topics__in=self.topics)
+            context['related_pages'] = get_related_pages(
+                self, self.publication_related_links.all(), queryset)
+        elif related_options == 'Country':
+            queryset = PublicationPage.objects.filter(page_countries__in=self.page_countries)
+            context['related_pages'] = get_related_pages(
+                self, self.publication_related_links.all(), queryset)
+        else:
+            context['related_pages'] = get_related_pages(
+                self, self.publication_related_links.all(), PublicationPage.objects)
```

Comment:
You have this almost exactly repeated in like 5 different places ... Use a function and add it in the `RelatedLinksMixin`

```
def get_related_pages(self, option='Manual', PageClass):
     ....
```

Usage

`self.get_related_pages(related_options, PublicationPage)` - be sure to replace PublicationPage with all the other page types as required.

Reply:
Great, thanks